File: front-end/src/app/service-models/app/v1/app-users/client-employee-user-counts-response-s-m.ts
import { CoinManagementServiceModelBase } from '../../base/coin-management-service-model-base';

export class ClientEmployeeUserCountsResponseSM extends CoinManagementServiceModelBase<number> {
    allEmployeeCount!: number;
    adminEmployeeCount!: number;
    simpleEmployeeCount!: number;
    activeEmployeeCount!: number;
    resignedEmployeeCount!: number;
    suspendedEmployeeUserCount!: number;
}

File: backend/payroll-backend/Components/Siffrum.Web.Payroll.ServiceModels/v1/AppUsers/ClientEmployeeUserCountsResponseSM.cs
using Siffrum.Web.Payroll.ServiceModels.Base;

namespace Siffrum.Web.Payroll.ServiceModels.v1.AppUsers
{
    public class ClientEmployeeUserCountsResponseSM : SiffrumPayrollServiceModelBase<int>
    {
        public int AllEmployeeCount { get; set; }
        public int AdminEmployeeCount { get; set; }
        public int SimpleEmployeeCount { get; set; }
        public int ActiveEmployeeCount { get; set; }
        public int ResignedEmployeeCount { get; set; }
        public int SuspendedEmployeeUserCount { get; set; }
    }
}
